app/Jobs/AutomateSessionEntries.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Models\Event;
+use App\Models\Session;
+use App\Models\Entry;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Carbon\Carbon;
+
+//Remove ShouldBeUnique if we partition the sessions to be handled
+class AutomateSessionEntries implements ShouldQueue, ShouldBeUnique
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        //Grab all existing live/past sessions in DB and load characters
+        $sessions = Session::where('start_time', '<=', Carbon::now())
+            ->where('start_time', '>', Carbon::now()->subMinutes(15))
+            ->with('characters')
+            ->get();
+
+        //entry creation input arrays for all characters & dm in $sessions
+        $input_arrays = collect([]);
+        foreach ($sessions as $session) {
+            foreach ($session->characters as $character) {
+                $input_arrays->push(collect([
```

Comment:
this call to `collect(...)` adds unnecessary overhead, you can just remove it

Reply:
I made calls to `collect(...)` because otherwise inputArrays is treated as a regular array and won't let me use collection functions like pluck. It seems to work fine if I only use it to initialize the empty collection. Within the loop you're right it doesn't seem to be necessary. Does that make sense or is there a better way to do that?